uint32 dlo_id # dlo index
SegmentState[] states